packages/realm/src/Listeners.ts

```diff
@@ -54,3 +56,43 @@ export class Listeners<CallbackType> {
     this.listeners.clear();
   }
 }
+
+/** @internal */
+type BindingListenerCallback = (r: Realm, name: string, schema?: Realm.ObjectSchema[]) => void;
+/** @internal */
+export class BindingListeners {
+  /**
+   * Keeps tracked of registered listener callbacks for Realm class notifications.
+   */
+  constructor(private name: string) {}
```

Comment:
I'd suggest a name closer to the semantics.

```suggestion
  constructor(private eventName: string) {}
```

Since this is passed to the callback as an argument and it has only three applicable values, I'd suggest using a string-based enum for this.